### PR TITLE
Asset-saver refactors

### LINKED ISSUE #1361: incorrect date in filepath
The date in filename appears to use UTC, but time is correct localtime (though no DST detected). 

Example using the CLI to write to html, format as `<site>_<date>_<time>.html`

```
...
html output written to www.kogan.com_2016-12-29_09-59-05.report.html

simon@delaware:~$ date
Friday 30 December  10:59:56 AEDT 2016
```

### PR DESCRIPTION
**Pull request by paulirish**
* Improved `getFilenamePrefix`.
  * Using `new URL()` for the hostname.
  * Fixes #1361 where inconsistent timezone used for date vs time
  * Added tests for this
* Changed parameters of `saveAssets`, `prepareAssets`, and `getFilenamePrefix`
  * Generally they all take the ending `results` object rather than the starting `options` object. So the URL is the finalUrl, rather than provided one. This lets use use the generatedTime, as before, we were not.
* Removed `filterForSize`. Unnecessary since the new FMP landed.

### CODE REVIEW

File: lighthouse-core/lib/asset-saver.js [hunk 1]
```diff
@@ -18,32 +18,36 @@
...
 }
 
-// Some trace events are particularly large, and not only consume a LOT of disk
-// space, but also cause problems for the JSON stringifier. For simplicity, we exclude them
-function filterForSize(traceData) {
-  return traceData.filter(e => e.name !== 'LayoutTree');
-}
-
-function screenshotDump(options, screenshots) {
+function screenshotDump(screenshots, results) {
```
Comment (ebidel) [inline lighthouse-core/lib/asset-saver.js]: can you doc this up while you're here?

File: lighthouse-core/lib/asset-saver.js [hunk 2]
```diff
@@ -87,11 +91,12 @@ function saveArtifacts(artifacts, filename) {
 
 /**
  * Filter traces and extract screenshots to prepare for saving.
- * @param {!Object} options
- * @param {!Artifacts} artifacts
+
+ * @param {!Artifacts} artifactsFilename
```
Comment (ebidel) [inline lighthouse-core/lib/asset-saver.js]: artifactsFilename  -> artifacts

File: lighthouse-core/test/lib/asset-saver-test.js [hunk 3]
```diff
@@ -37,15 +55,15 @@ describe('asset-saver helper', () => {
...
       requestScreenshots: () => Promise.resolve([]),
     };
-    return assetSaver.prepareAssets(options, artifacts).then(assets => {
+    return assetSaver.prepareAssets(artifacts, options).then(assets => {
       assert.ok(/<!doctype/gim.test(assets[0].html));
     });
   });
 
-  describe('saves files to disk with real filenames', function() {
+  describe('saves files.', function() {
```
Comment (ebidel) [inline lighthouse-core/test/lib/asset-saver-test.js]: "saves files." FULL. STOP.

**Comment by wardpeet** [comment]
just a question why choose finalUrl above options url?
Wouldn't this make it more confusing?
Lets say I have a redirect like this
`example.com/splash` => `example.com/internal/is/long?page=splash`

**Comment by ebidel** [comment]
IMO, using `finalURL` is an improvement. We use the final url in the report (CLI and HTML). Naming the file with the same URL will be less ambiguous.

**Comment by brendankenny** [comment]
👍 on `finalURL`.

> Lets say I have a redirect like this `example.com/splash` => `example.com/internal/is/long?page=splash`

In this case the filename wouldn't be affected, because `getFilenamePrefix()` only uses the `hostname`.

It *would* pickup a redirect from `example.com` to `m.example.com`, though, which I think is an improvement in clarity.

**Comment by wardpeet** [comment]
ah if it only uses hostname it indeed much better! 😄 👍

File: lighthouse-core/lib/asset-saver.js [hunk 4]
```diff
@@ -19,31 +19,41 @@
...
+ */
+function getFilenamePrefix(results) {
+  const hostname = new URL(results.url).hostname;
+  const date = (results.generatedTime && new Date(results.generatedTime)) || new Date();
+
+  const timeStr = date.toLocaleTimeString('en-US', {hour12: false});
+  const dateParts = date.toLocaleDateString('en-US', {
+    year: 'numeric', month: '2-digit', day: '2-digit'
+  }).split('/');
+  dateParts.unshift(dateParts.pop());
```
Comment (brendankenny) [inline lighthouse-core/lib/asset-saver.js]: lol, nice

File: lighthouse-core/lib/asset-saver.js [hunk 5]
```diff
@@ -77,22 +87,27 @@ img {
   `;
 }
 
+/**
+ * Save entire artifacts object to a single stringified file
+ * @param {!Artifacts} artifacts
+ * @param {!artifactsFilename} artifactsFilename
```
Comment (brendankenny) [inline lighthouse-core/lib/asset-saver.js]: isn't this `@param {string} artifactsFilename`?

**Pull request merged**